apps/src/code-studio/components/ShareAllowedDialog.jsx

```diff
@@ -25,11 +25,28 @@ import Button from '../../templates/Button';
 import Spinner from '@cdo/apps/code-studio/pd/components/spinner';
 import defaultThumbnail from '@cdo/static/projects/project_default.png';
 import fontConstants from '@cdo/apps/fontConstants';
+import analyticsReporter from '@cdo/apps/lib/util/AnalyticsReporter';
+import {EVENTS} from '@cdo/apps/lib/util/AnalyticsConstants';
+
+const SHARE_TYPE_EVENT = {
+  'open': EVENTS.SHARING_DIALOG_OPEN,
+  'copy': EVENTS.SHARING_LINK_COPY,
+  'publish': EVENTS.SHARING_PUBLISH,
+  'facebook': EVENTS.SHARING_FB,
+  'twitter': EVENTS.SHARING_TWITTER,
+  'send-to-phone': EVENTS.SHARING_LINK_SEND_TO_PHONE,
+  'close': EVENTS.SHARING_CLOSE_ESCAPE,
+};
 
-function recordShare(type) {
+function recordShare(type, appType) {
   if (!window.dashboard) {
     return;
   }
+  if (SHARE_TYPE_EVENT[type]) {
+    analyticsReporter.sendEvent(SHARE_TYPE_EVENT[type], {
+      lab_type: appType,
```

Comment:
Any value to including project ID as well?

Reply:
Based on [jira ticket](https://codedotorg.atlassian.net/browse/LABS-496) Product would like to 'beter understand usage of the share feature' - unsure if project id is needed fo this.

@samantha-code - lmk what you think.

Reply:
What exactly is project ID? Unique identifier per project?

Reply:
Yes - that's correct. It was included in the firehose logging we're moving [here](https://github.com/code-dot-org/code-dot-org/blob/6bd144722f7d33d36ece8749d9483a352204c0dd/apps/src/code-studio/components/ShareAllowedDialog.jsx#L39).

Reply:
I dont think it could hurt to add! It will be interesting to see if one project is being shared more than others

Reply:
Thanks! And just to clarify, this is the project `channel_id` which can be decrypted to get the `storage_id` and `project_id`.